File: DotNet/Messaging/Storage/ByteArrayAttachment.cs

namespace Shastra.Hydra.Messaging.Storage
{
    public class ByteArrayAttachment : Attachment
    {
        private const string DefaultContentType = "application/octet-stream";

        public byte[] Data { get; private set; }
        public string ContentType { get; private set; }

        public ByteArrayAttachment(string name, byte[] data, string contentType = DefaultContentType) : base(name)
        {
            Data = data;
            ContentType = contentType;
        }    
    }
}

File: Java/hydra/src/main/java/uk/co/shastra/hydra/messaging/attachments/ByteArrayAttachment.java
package uk.co.shastra.hydra.messaging.attachments;

import org.apache.http.entity.ContentType;
import org.apache.http.entity.mime.content.ByteArrayBody;
import org.apache.http.entity.mime.content.ContentBody;

public class ByteArrayAttachment extends Attachment {
	
    private static final String DefaultContentType = "application/octet-stream";

    private byte[] data;

    public byte[] getData() { return data; }

    public ByteArrayAttachment(String name, byte[] data) { this(name, data, DefaultContentType); }
    
	public ByteArrayAttachment(String name, byte[] data, String contentType)
    {
    	super(name, contentType);
        this.data = data;
    }

	@Override
	public int dataLength() { return data.length; }

	@Override
	public ContentBody toContentBody() { return new ByteArrayBody(data, ContentType.create(getContentType()), ""); }

}
